apps/emqx/include/logger.hrl

```diff
@@ -75,12 +75,18 @@
 %% Internal macro
 -define(_DO_TRACE(Tag, Msg, Meta),
     case persistent_term:get(?TRACE_FILTER, []) of
-        [] -> ok;
+        [] ->
+            ok;
         %% We can't bind filter list to a variable because we pollute the calling scope with it.
         %% We also don't want to wrap the macro body in a fun
         %% because this adds overhead to the happy path.
         %% So evaluate `persistent_term:get` twice.
-        _ -> emqx_trace:log(persistent_term:get(?TRACE_FILTER, []), Msg, (Meta)#{trace_tag => Tag})
+        _ ->
+            emqx_trace:log(
+                persistent_term:get(?TRACE_FILTER, []),
+                Msg,
+                maps:merge(Meta, #{trace_tag => Tag})
```

Comment:
if this change is due to otp 27, maybe revert, otherwise it will conflict master branch

Reply:
yes, due to silence compiler warning.
This has been changed in PR#14514, I will revert it.